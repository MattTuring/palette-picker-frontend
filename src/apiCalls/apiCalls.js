export const getProjects = async () => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/projects';
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error('There was an error getting projects.');
  }
  const projects = await response.json();
  return projects
}

export const getPalettes = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}/palettes`
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error('There was an error getting palettes.');
  }
  const palettes = await response.json();
  return palettes
}

export const postProject = async (project) => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/projects';
  const options = {
    method: 'POST',
    body: JSON.stringify(project),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error posting project.');
  }

  const result = await response.json();
  return result;
}

export const postPalette = async (palette) => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/palettes';
  const options = {
    method: 'POST',
    body: JSON.stringify(palette),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error posting palette.');
  }

  const result = await response.json();
  return result;
}

export const updateProject = async (project, id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}`;
  const options = {
    method: 'PUT',
    body: JSON.stringify(project),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error updating project.');
  }

  const result = await response.json();
  return result;
}

export const updatePalette = async (palette, id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/palettes/${id}`;
  const options = {
    method: 'PUT',
    body: JSON.stringify(palette),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error updating palette.');
  }

  const result = await response.json();
  return result;
}

export const deleteProject = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}`;
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error deleting project.');
  }

  const result = await response.json();
  return result;
}

export const deletePalette = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/palettes/${id}`;
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error deleting palette.');
  }

  const result = await response.json();
  return result;
}
