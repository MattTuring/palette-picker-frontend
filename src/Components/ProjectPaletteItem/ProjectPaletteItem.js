
import React, {useState, useEffect} from 'react';
import { deletePalette, updatePalette } from '../../apiCalls/apiCalls';
import trash from '../../trash.png';
import edit from '../../edit.png';

const ProjectPaletteItem = ({palette, palettes, setPalettes}) => {

  const [name, setName] = useState('');
  useEffect(() => setName(palette.name), []);
  const removeItem = async () => {
    try {
      const newPallete = palettes.filter(pal => pal.id !== palette.id);
      setPalettes(newPallete);
      await deletePalette(palette.id);
    }
    catch (error) {
      console.log(error);
    }
  }


  let [newPaletteName, setPaletteNewName] = useState('');
  let [showPaletteNewName, setPaletteShowNewName] = useState(false);

  return (
    <>
      <p style={{display: 'none'}}>{palette.project_id}</p>
      <p className='palette-area'>{name}
        <div className='color-div' style={{backgroundColor: palette.color1}}></div>
        <div className='color-div' style={{backgroundColor: palette.color2}}></div>
        <div className='color-div' style={{backgroundColor: palette.color3}}></div>
        <div className='color-div' style={{backgroundColor: palette.color4}}></div>
        <div className='color-div' style={{backgroundColor: palette.color5}}></div>
        <img src={trash} onClick={removeItem} className="delete" alt='delete'/>

        <img src={edit} onClick={() => setPaletteShowNewName(!showPaletteNewName)} className="edit" alt='edit'/>
        {showPaletteNewName && <><input onChange={(event) => setPaletteNewName(event.target.value)} type='text' placeholder='New Palette Name'/><button onClick={() => { updatePalette({name: newPaletteName},palette.id); setName(newPaletteName)}}>Update Name</button></>}

      </p>
    </>
  )
};

export default ProjectPaletteItem;
