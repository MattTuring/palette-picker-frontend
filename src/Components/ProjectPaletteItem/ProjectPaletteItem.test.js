import React from 'react';
import { shallow } from 'enzyme';
import ProjectPaletteItem from './ProjectPaletteItem';
import { deletePalette } from '../../apiCalls/apiCalls';

jest.mock('../../apiCalls/apiCalls');

describe('ProjectPaletteItem', () => {
  const setPalettes = jest.fn();
  const mockProps = {
    palette: {
      id: 1,
      project_id: 1,
      color1: '#000000',
      color2: '#ffffff',
      color3: '#000000',
      color4: '#ffffff',
      color5: '#000000',
    }, palettes: [{
      id: 1,
      project_id: 1,
      color1: '#000000',
      color2: '#ffffff',
      color3: '#000000',
      color4: '#ffffff',
      color5: '#000000',
    }],
    setPalettes
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ProjectPaletteItem {...mockProps} />
    );
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setPalettes after click on remove button', () => {
    wrapper.find('.delete').simulate('click');
    expect(setPalettes).toHaveBeenCalledWith([]);
  });

  it('should call deletePalette after click on remove button', () => {
    wrapper.find('.delete').simulate('click');
    expect(deletePalette).toHaveBeenCalledWith(1);
  });
});
