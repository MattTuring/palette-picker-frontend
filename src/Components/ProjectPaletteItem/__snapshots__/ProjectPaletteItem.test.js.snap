// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPaletteItem should match the snapshot with all data passed in correctly 1`] = `
<Fragment>
  <p
    style={
      Object {
        "display": "none",
      }
    }
  >
    1
  </p>
  <p
    className="palette-area"
  >
    <div
      className="color-div"
      style={
        Object {
          "backgroundColor": "#000000",
        }
      }
    />
    <div
      className="color-div"
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    />
    <div
      className="color-div"
      style={
        Object {
          "backgroundColor": "#000000",
        }
      }
    />
    <div
      className="color-div"
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    />
    <div
      className="color-div"
      style={
        Object {
          "backgroundColor": "#000000",
        }
      }
    />
    <img
      alt="delete"
      className="delete"
      onClick={[Function]}
      src="trash.png"
    />
    <img
      alt="edit"
      className="edit"
      src="edit.png"
    />
  </p>
</Fragment>
`;
